### PR TITLE
Decouple RestrictedZoneChecker from ProductInterface

### PR DESCRIPTION
**Pull request by videni**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Related tickets | fixes #X, partially #Y, mentioned in #Z |
| License | MIT |
| Doc PR | Sylius/Sylius-Docs#X |

### CODE REVIEW

File: src/Sylius/Component/Core/Model/RestrictedZoneInterface.php [hunk 1]
```diff
@@ -0,0 +1,22 @@
+<?php
+
+namespace Sylius\Component\Core\Model;
+
+use Sylius\Component\Addressing\Model\ZoneInterface;
+
+/**
+ * @author   vidy   <videni@foxmail.com>
+ */
+
```
Comment (Zales0123) [inline src/Sylius/Component/Core/Model/RestrictedZoneInterface.php]: Redundant blank line.

File: src/Sylius/Component/Core/Model/RestrictedZoneInterface.php [hunk 2]
```diff
@@ -0,0 +1,22 @@
+<?php
+
+namespace Sylius\Component\Core\Model;
+
+use Sylius\Component\Addressing\Model\ZoneInterface;
+
+/**
+ * @author   vidy   <videni@foxmail.com>
```
Comment (Zales0123) [inline src/Sylius/Component/Core/Model/RestrictedZoneInterface.php]: Some wrong formatting here.

File: src/Sylius/Component/Core/Model/RestrictedZoneInterface.php [hunk 3]
```diff
@@ -0,0 +1,22 @@
+<?php
+
```
Comment (Zales0123) [inline src/Sylius/Component/Core/Model/RestrictedZoneInterface.php]: Missing license block.

File: src/Sylius/Component/Core/Model/RestrictedZoneInterface.php [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    /**
+     * @return ZoneInterface
+     */
+    public function getRestrictedZone();
+
+    /**
+     * @param ZoneInterface $zone
+     */
+    public function setRestrictedZone(ZoneInterface $zone = null);
+}
```
Comment (Zales0123) [inline src/Sylius/Component/Core/Model/RestrictedZoneInterface.php]: Missing blank line at the end of file.

**Comment by adamelso** [comment]
`RestrictedZoneInterface` implies that the implementation is an instance of `Zone`, rather than it being subject to restrictions for zones.

Name it `RestrictedZoneSubjectInterface` - that way it is also consistent with the rest of Sylius.

(You could name it `RestrictedInterface` but that might imply a restriction by something else and not just a zone.)

**Comment by pamil** [comment]
@adamelso :+1:

File: src/Sylius/Component/Core/Model/RestrictedZoneInterface.php [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+interface RestrictedZoneInterface
+{
+    /**
+     * @return ZoneInterface
+     */
+    public function getRestrictedZone();
+
+    /**
+     * @param ZoneInterface $zone
```
Comment (pamil) [inline src/Sylius/Component/Core/Model/RestrictedZoneInterface.php]: `ZoneInterface|null`

File: src/Sylius/Component/Core/Model/RestrictedZoneInterface.php [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+use Sylius\Component\Addressing\Model\ZoneInterface;
+
+/**
+ * @author   vidy   <videni@foxmail.com>
+ */
+
+interface RestrictedZoneInterface
+{
+    /**
+     * @return ZoneInterface
```
Comment (pamil) [inline src/Sylius/Component/Core/Model/RestrictedZoneInterface.php]: `ZoneInterface|null`

**Comment by pjedrzejewski** [comment]
I was actually considering removing this feature until we test it properly with Behats and consider all different scenarios. Current implementation is quite simple and if someone really needs that feature it can be implemented with easy in the project. Every feature means more possible bugs and we want to stabilize Sylius.

**Comment by videni** [comment]
I agree with @pjedrzejewski, removing this feature doesn't hurt anything, but makes things simple.

**Comment by Zales0123** [comment]
It can be closed, as `RestrictedZoneChecker` was removed in https://github.com/Sylius/Sylius/pull/5198

**Pull request closed without merging**